Evaluator/TA/ai_evaluator/ai.py

```diff
@@ -135,56 +153,75 @@ async def ohlcv_callback(self, exchange: str, exchange_id: str,
+                    self.logger.error(f"Too many requests: {e}")
+                except services_errors.UnavailableInBacktestingError:
+                    # error already logged error for backtesting in use_backtesting_init_timeout
+                    pass
+                except evaluators_errors.UnavailableEvaluatorError as e:
+                    self.logger.exception(e, True, f"Evaluation error: {e}")
+                except tulipy.lib.InvalidOptionError as e:
+                    self.logger.warning(
+                        f"Error when computing {self.indicator} on {self.period} period with {len(candle_data)} "
+                        f"candles: {e}"
+                    )
+                    self.logger.exception(e, False)
+            else:
+                self.logger.debug(f"Ignored {time_frame} time frame as the shorted allowed time frame is "
+                                  f"{self.min_allowed_timeframe}")
+            await self.evaluation_completed(cryptocurrency, symbol, time_frame,
+                                            eval_time=evaluators_util.get_eval_time(full_candle=candle,
+                                                                                    time_frame=time_frame))
 
-    async def ask_gpt(self, preprompt, inputs, symbol, time_frame) -> str:
+    async def ask_gpt(self, preprompt, inputs, symbol, time_frame, candle_time) -> str:
         try:
-            service = await services_api.get_service(gpt_service.GPTService, self.is_backtesting)
+            service = await services_api.get_service(
+                gpt_service.GPTService,
+                self.is_backtesting,
+                {} if self.is_backtesting else self.services_config
+            )
             resp = await service.get_chat_completion(
                 [
                     service.create_message("system", preprompt),
                     service.create_message("user", inputs),
                 ],
-                model=self.gpt_model if self.enable_model_selector else None
+                model=self.gpt_model if self.enable_model_selector else None,
+                exchange=self.exchange_name,
+                symbol=symbol,
+                time_frame=time_frame,
+                version=self.get_version(),
+                candle_open_time=candle_time,
+                use_stored_signals=self.is_backtesting
             )
             self.logger.info(f"GPT's answer is '{resp}' for {symbol} on {time_frame} with input: {inputs}")
             return resp
         except services_errors.CreationError as err:
             raise evaluators_errors.UnavailableEvaluatorError(f"Impossible to get ChatGPT prediction: {err}") from err
+        except Exception as err:
+            print(err)
+
+    def get_version(self):
+        # later on, identify by its specs
+        # return f"{self.gpt_model}-{self.source}-{self.indicator}-{self.period}-{self.GLOBAL_VERSION}"
+        return "0.0.0"
```

Comment:
I don't understand why the version should contains all these data and not just a number (here `GLOBAL_VERISON`)